File: new-post.rb
# Ruby script to generate the next file for writing today's post in

require 'date'

POSTS_DIR = "_posts"

file_name = "#{Date.today.iso8601}.md"
file_path = "#{POSTS_DIR}/#{file_name}"

puts "CREATED #{file_name}"

filout = File.open file_path, "w"

filout << "---
layout: post
comments: true
title:
categories:
---\n\n"

filout.close()

puts "TEMPLATE inserted in #{file_path}"

File: new-post.pl
use strict;
use warnings;

use POSIX qw(strftime);

if (@ARGV < 1) {
    die "You must provide the post title as an argument";
}

my $stub = $ARGV[0];
if ($stub !~ m/^[\w-]+$/i) {
    die "Provided blog post sub must match [a-z-]+";
}

my $file_name = "_posts/";
$file_name .= strftime "%Y-%m-%d-", localtime;
$file_name .= $stub;
$file_name .= ".md";

my $post_template = '---
layout: post
comments: true
title: 
categories:
---';

open (my $pipe, ">", $file_name);
print $pipe $post_template;
close($pipe);

print "Wrote template to file\n$file_name\n";
